src/App.tsx

```diff
@@ -1,27 +1,65 @@
 import React from 'react';
-import './App.scss';
-
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  // eslint-disable-next-line
-  console.log(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+import { Clock } from './components/Clock';
+
+interface State {
+  isClockVisible: boolean,
+  clockName: number,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    clockName: 1,
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <p>{`Clock Name: ${clockName}`}</p>
+        <p>
+          <p>Local time:</p>
+          {isClockVisible && (
+            <Clock from={clockName} />
+          )}
+        </p>
+        <button
+          type="button"
+          onClick={() => {
+            this.setState({
+              isClockVisible: true,
+            });
+          }}
+        >
+          Start
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            this.setState({
+              isClockVisible: false,
+            });
+          }}
+        >
+          Stop
```

Comment:
it says stop, but it does not stop the timer. it only hides it

Reply:
clearInterval sets off on `componentWillUnmount`
```
componentWillUnmount() {
    if (this.timerId) {
      clearInterval(this.timerId);
    }
  }
```